src/DotNetWorker.Core/Hosting/ServiceCollectionExtensions.cs

```diff
@@ -81,7 +81,7 @@ public static IFunctionsWorkerApplicationBuilder AddFunctionsWorkerCore(this ISe
                     }
                 });
 
-            services.AddSingleton<ILoggerProvider, WorkerLoggerProvider>();
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<ILoggerProvider, WorkerLoggerProvider>());
```

Comment:
Non-blocking question just for my own understanding, but why is this preferable to using `TryAddSingleton`?

Reply:
Depends on what the registered service is being used for. For `ILoggerProvider`, there will be many registered so we don't want to use `TryAddSingleton()` as it'd just flat-out skip this if anyone else had registered an `ILoggerProvider`.

Instead, `TryAddEnumerable()` checks if this specific implementation type has already been registered for this service type. Only then will it skip. This allows other types to be registered alongside it, but will prevent this specific duplication.